client/src/fabric/features.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2019. Beautiful Code BV, Rotterdam, Netherlands
+ * Licensed under GNU GENERAL PUBLIC LICENSE Version 3.
+ */
+
+import { BehaviorSubject } from "rxjs"
+
+import { IFabricEngine, IntervalRole, PhysicsFeature } from "./fabric-engine"
+import { FabricInstance } from "./fabric-kernel"
+
+export interface IFeatureName {
+    physicsFeature?: PhysicsFeature
+    intervalRole?: IntervalRole
+}
+
+function nameLabel(feature: IFeatureName): string {
+    if (feature.physicsFeature !== undefined) {
+        return PhysicsFeature[feature.physicsFeature]
+    }
+    if (feature.intervalRole !== undefined) {
+        return IntervalRole[feature.intervalRole]
+    }
+    return "?"
+}
+
+export interface IFeature {
+    label: string
+    name: IFeatureName
+    isPhysics: boolean
+    setFactor: (factor: number) => void
+    factor$: BehaviorSubject<number>
+    defaultValue: number
+}
+
+export function applyFeatureToInstance(instance: FabricInstance):  (feature: IFeature) => void {
+    return feature => {
+        const intervalRole = feature.name.intervalRole
+        if (intervalRole === undefined) {
+            throw new Error()
+        }
+        const factor = feature.factor$.getValue()
+        instance.setRoleLength(intervalRole, factor)
+    }
+}
+
+export function applyFeatureToEngine(engine: IFabricEngine): (feature: IFeature) => void {
+    return feature => {
+        const globalFeature = feature.name.physicsFeature
+        if (globalFeature === undefined) {
+            throw new Error()
+        }
+        const factor = feature.factor$.getValue()
+        engine.setPhysicsFeature(globalFeature, factor)
+    }
+}
+
+export interface IFeatureStorage {
+    getFeature: (label: string, defaultValue: number) => number
+    setFeature: (label: string, factor: number) => void
+}
+
+export function enumToFeatureArray(enumObject: object, physics: boolean, storage: IFeatureStorage): IFeature[] {
+    return Object.keys(enumObject)
+        .filter(key => key.length > 1)
```

Comment:
Is this to filter out the reverse keys e.g. "1" -> "SomeEnumValue"? If so, this'll break once you add enum value #10

Reply:
Yes.  Fixed that.